spot-electron/src/bt-beacon/daemon/darwin/bluetoothd.js

```diff
@@ -0,0 +1,159 @@
+const XpcConnection = require('xpc-connect');
+
+const { logger } = require('../../../logger');
+
+const DEVICE_STATES = [ 'Unknown', 'Resetting', 'Unsupported', 'Unauthorized', 'Powered Off', 'Powered On' ];
+
+/**
+ * Implements a platform specific daemon to interact with the BLE hardware in the computer/device.
+ */
+class BluetoothD {
+    advertisingStatusUpdatedPromise;
+
+    /**
+     * Instantiates a new instance.
+     */
+    constructor() {
+        this.xpcConnection = new XpcConnection('com.apple.bluetoothd');
+
+        this.xpcConnection.on('error', message => {
+            logger.error('XpcError', message);
+        });
+
+        this.xpcConnection.on('event', event => {
+            const messageArguments = event.kCBMsgArgs;
+
+            switch (event.kCBMsgId) {
+            case 4:
+                // Device status update
+                this.setStatus(messageArguments.kCBMsgArgState);
+                break;
+            case 28:
+                // Beacon activation result
+                this.setAdvertisingStatus(!messageArguments.kCBMsgArgResult);
+                break;
+            case 29:
+                // Beacon deactivation result
+                this.setAdvertisingStatus(Boolean(messageArguments.kCBMsgArgResult));
+                break;
+            case 38:
+                // Device ready
+                this.readyPromise && this.readyPromise();
+                break;
+            case 67:
+                // Connection request. We don't need this.
+                break;
+            default:
+                logger.warn('Unknown XpcEvent', event);
+            }
+        });
+    }
+
+    /**
+     * Connects to the BT device (opens XPC channel).
+     *
+     * @returns {Promise<void>}
+     */
+    connect() {
+        return new Promise(resolve => {
+            this.readyPromise = resolve;
+            this.xpcConnection.setup();
+            this.sendMessage(1, {
+                kCBMsgArgName: `jitsi-bt-beacon-${(new Date()).getTime()}`,
+                kCBMsgArgOptions: {
+                    kCBInitOptionShowPowerAlert: 1
+                },
+                kCBMsgArgType: 1,
+                kCBMsgArgVersion: 20161219
```

Comment:
There are a lot of magical numbers in this file. Would someone working on bluetooth be able to quickly get context on what these numbers mean?

Reply:
Not really, these are API constants. I don't know what they mean and frankly don't even care :)

Reply:
Okay. Maybe when doing the windows work I'll have to look. I'll use constants for the numbers if I have to figure out what they mean.